addon/models/resource.js

```diff
@@ -2,16 +2,45 @@ import Ember from 'ember';
 import { pluralize } from 'ember-inflector';
 
 const Resource = Ember.Object.extend({
+
+  /*
+    Defaults for a JSON API Resource object
+  */
+  id: null,
   type: null,
-  links: {},
-  attributes: {},
-  relationships: {},
+  links: null,
+  attributes: null,
+  relationships: null,
+
+  /*
+    init a JSON API Resource object with default as empty objects on the instance
+  */
+  init() {
+    const attrs = this.getProperties('links', 'attributes', 'relationships');
+    this.setProperties({
+      'links': attrs.links || {},
+      'attributes': attrs.attributes || {},
```

Comment:
Why set 'attributes' to `attrs.attributes` if it is present instead of always creating a new object? Doesn't that expose the same problem to subclasses if they override 'attributes'? Or is the goal to provide that flexibility?